django_ai_assistant/helpers/assistants.py

```diff
@@ -422,7 +450,36 @@ class AgentState(TypedDict):
             output: str
 
         def setup(state: AgentState):
-            return {"messages": [SystemMessage(content=self.get_instructions())]}
+            messages = [SystemMessage(content=self.get_instructions())]
+
+            if self.structured_output:
+                schema = None
+
+                # If Pydantic
+                if inspect.isclass(self.structured_output) and issubclass(
+                    self.structured_output, BaseModel
+                ):
+                    schema = json.dumps(self.structured_output.model_json_schema())
+
+                schema_information = ""
+                if schema:
+                    schema_information = f"JSON will have the following schema:\n\n{schema}\n\n"
+
+                # The assistant won't have access to the schema of the structured output before
+                # the last step of the chat. This message gives visibility about what fields the
+                # response should have so it can gather the necessary information by using tools.
+                messages.append(
+                    SystemMessage(
+                        content=(
+                            "In the last step of this chat you will be asked to respond in JSON. "
+                            + schema_information
+                            + "Gather information using tools. "
+                            "Don't generate JSON until you are explicitly told to. "
```

Comment:
When the assistant is used in 100% JSON mode, maybe that prompt confuses the LLM?

Reply:
I haven't had this issue in my tests but I don't expect it to be a problem since when the the llm will be explicitly instructed to return JSON. I'm 100% it won't be a problem with ChatGPT with strict "json_schema" mode because it ensures the response is JSON, for other models it might cause some issues, I'm not certain of what would be the impact.